pallets/ajuna-board/src/lib.rs

```diff
@@ -243,4 +287,31 @@ impl<T: Config> Pallet<T> {
 			None => Seed::<T>::kill(),
 		}
 	}
+
+	fn declare_winner(
+		board_id: &T::BoardId,
+		winner: &<<T as Config>::Game as TurnBasedGame>::Player,
+		board_state: &<<T as Config>::Game as TurnBasedGame>::State,
+	) {
+		// Cache result in storage, this would be cleared on `flush_winner`
+		BoardWinners::<T>::insert(board_id, winner.clone());
+		BoardExpiries::<T>::remove(board_id);
+		Self::seed_for_next(board_state);
+		Self::deposit_event(Event::GameFinished { board_id: *board_id, winner: winner.clone() });
+	}
+
+	fn set_or_update_expiry(board_id: T::BoardId) {
```

Comment:
Could we rename this to:
```suggestion
	fn upsert_expiry(board_id: T::BoardId) {
```